quantmsio/core/duckdb.py

```
@@ -1,104 +1,528 @@
+
+        # Basic counts
+        stats.update(
+            self.query_to_df(
+                f"""
+            SELECT 
+                COUNT(*) as total_psms,
+                COUNT(DISTINCT "{PROTEIN_GROUP}") as total_proteins,
+                COUNT(DISTINCT "{MODIFIED_SEQUENCE}") as unique_peptides,
+                COUNT(DISTINCT "{RUN}") as total_runs
+            FROM report
+        """
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)